cmd/delete/cluster.go

```diff
@@ -61,22 +67,40 @@ func NewClusterCmd() *cobra.Command {
 				return ErrForceFlagNotSet
 			}
 
-			execx.Debug = debug
-
 			// Init paths.
 			homeDir, err := os.UserHomeDir()
 			if err != nil {
 				return fmt.Errorf("error while getting user home directory: %w", err)
 			}
 
-			minimalConf, err := yamlx.FromFileV3[config.Furyctl](furyctlPath)
+			// Init first half of collaborators.
+			client := netx.NewGoGetterClient()
+			executor := execx.NewStdExecutor()
+			distrodl := distribution.NewDownloader(client, debug)
```

Comment:
these should be injected, and an interface should be used as type hint in the constructor.
Seeing we did this in many places, we could probably open a dedicated card to refactor it if we deem it necessary.

Reply:
yep seems like a card is needed to plan this refactor